### PR TITLE
Throw NoSuchIcebergTableException instead of ValidationException in G…

### PR DESCRIPTION
**Pull request by ericlgoodman**
Currently, the Hive catalog is throwing a `NoSuchIcebergTableException` upon discovering that the table in question is not Iceberg.

This is most apparent in Spark, where currently `SparkSessionCatalog` in conjunction with `GlueCatalog` cannot read non-Iceberg tables despite that being the purpose of the `SparkSessionCatalog` (falling back to the built-in catalog)

### CODE REVIEW

File: aws/src/main/java/org/apache/iceberg/aws/glue/GlueToIcebergConverter.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 import org.apache.iceberg.BaseMetastoreTableOperations;
 import org.apache.iceberg.catalog.Namespace;
 import org.apache.iceberg.catalog.TableIdentifier;
-import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.exceptions.NoSuchIcebergTableException;import org.apache.iceberg.exceptions.ValidationException;
```
Comment (jackye1995) [inline aws/src/main/java/org/apache/iceberg/aws/glue/GlueToIcebergConverter.java:24]: nit: newline for import

File: aws/src/main/java/org/apache/iceberg/aws/glue/GlueToIcebergConverter.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ static TableIdentifier toTableId(Table table) {
    */
   static void validateTable(Table table, String fullName) {
     String tableType = table.parameters().get(BaseMetastoreTableOperations.TABLE_TYPE_PROP);
-    ValidationException.check(
+    NoSuchIcebergTableException.check(
```
Comment (jackye1995) [inline aws/src/main/java/org/apache/iceberg/aws/glue/GlueToIcebergConverter.java:48]: From the perspective of this util method, it does not make much sense to throw `NoSuchIcebergTableException`. I would propose just do a try-catch in `doRefresh` and throw the right exception there and add a few lines of documentation for why.

File: aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java [hunk 3]
```diff
@@ -194,6 +195,21 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
     }
   }
 
+  /**
+   * Validate the Glue table is Iceberg table by checking its parameters
```
Comment (jackye1995) [inline aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java:199]: can we update the javadoc here describing why `NoSuchIcebergTableException` is used? Also should add `@throws` line in the end

Comment (ericlgoodman) [reply aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java]: Updated

File: aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java [hunk 4]
```diff
@@ -194,6 +195,24 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
...
+  /**
+   * Validate the Glue table is Iceberg table by checking its parameters. We throw a {@link
+   * NoSuchIcebergTableException} here so that the catalog can decide whether to fail or potentially
+   * use the Spark built-in catalog instead.
+   *
+   * @param table glue table
+   * @param fullName full table name for logging
+   * @throws NoSuchIcebergTableException if the table is not an Iceberg table
+   */
+  static void checkIfTableIsIceberg(Table table, String fullName) {
```
Comment (jackye1995) [inline aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java:207]: I think we should not say anything Spark related in the core library, conceptually it is more like "if the table properties check does not pass, for Iceberg it is equivalent to not having a table in the catalog"

**Review by jackye1995 — APPROVED** [review]
looks good to me!

File: aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java [hunk 5]
```diff
@@ -194,6 +195,24 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
...
+  /**
+   * Validate the Glue table is Iceberg table by checking its parameters. If the table properties
+   * check does not pass, for Iceberg it is equivalent to not having a table in the catalog. We
+   * throw a {@link NoSuchIcebergTableException} in that case.
+   *
+   * @param table glue table
+   * @param fullName full table name for logging
+   * @throws NoSuchIcebergTableException if the table is not an Iceberg table
+   */
+  static void checkIfTableIsIceberg(Table table, String fullName) {
```
Comment (amogh-jahagirdar) [inline aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java:207]: Nonblocking nit: can we rename this to `validateIsIcebergTable`

**Review by amogh-jahagirdar — APPROVED** [review]
LGTM, just a naming nit

File: aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java (hunk 5 above)
Comment (ericlgoodman) [reply aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java:207]: Discussed offline - `validate` may imply `ValidationException`. We'll stick with `checkIfTableIsIceberg`

**Comment by jackye1995** [comment]
Thanks for the fix, merging!

**Pull request merged**